line-bot-api-client/src/test/java/com/linecorp/bot/client/LineMessagingClientImplIntegrationTest.java

```diff
@@ -50,20 +83,46 @@
     public static final URL TEST_RESOURCE = ClassLoader.getSystemResource("integration_test_settings.yml");
     private LineMessagingClient target;
     private String userId;
+    private IntegrationTestSettings settings;
 
     @Before
     public void setUp() throws IOException {
+        // Do not run all test cases in this class when src/test/resources/integration_test_settings.yml doesn't
+        // exist.
         Assume.assumeTrue(TEST_RESOURCE != null);
 
-        final Map<?, ?> map = new ObjectMapper()
-                .convertValue(new Yaml().load(TEST_RESOURCE.openStream()), Map.class);
+        settings = new ObjectMapper()
+                .registerModule(new ParameterNamesModule())
+                .convertValue(new Yaml().load(TEST_RESOURCE.openStream()), IntegrationTestSettings.class);
 
         target = LineMessagingClient
-                .builder((String) map.get("token"))
-                .apiEndPoint((String) map.get("endpoint"))
+                .builder(settings.token)
+                .apiEndPoint(settings.endpoint)
                 .build();
 
-        userId = (String) map.get("userId");
+        userId = settings.userId;
+    }
+
+    @Value
+    public static final class IntegrationTestSettings {
+        private final String token;
+        private final String endpoint;
+        private final String userId;
+        private final List<String> audienceIfas;
+        private final String retargetingRequestId;
+
+        @JsonCreator
+        public IntegrationTestSettings(@JsonProperty("token") String token,
```

Comment:
I think Lombok's `@Builder` pattern is better.

Reply:
ok